### ISSUE TITLE
Alphabetical and '@key'

### ISSUE THREAD

**Issue by dillonredding** [stoplightio/spectral#730]
## Describe the bug

I may be doing it wrong, but the `alphabetical` function doesn't appear to work with `'@key'`. Here's the rule I'm trying to create:

```yaml
  response-order:
    message: Responses should be in alphabetical order
    recommended: true
    given: $.paths.*.*
    then:
      field: responses
      function: alphabetical
      functionOptions:
        keyedBy: '@key'
```

I've also tried this form:

```yaml
  response-order:
    message: Responses should be in alphabetical order
    recommended: true
    given: $.paths.*.*.responses
    then:
      field: '@key'
      function: alphabetical
```

## To Reproduce

Given this OpenAPI document

```yaml
openapi: 3.0.2
info:
  title: Test Spec
  version: 0.0.0
paths:
  /foo:
    get:
      operationId: get-foo
      responses:
        '400':
          description: ''
        '200':
          description: ''
```

Run this CLI command

```yaml
spectral lint test-spec.yaml -r my-rules.yaml
```

## Expected behavior

The lint output should contain the following:

```
 9:17  warning  response-order    Responses should be in alphabetical order
```

## Environment

 - Library version: 4.2.0
 - OS: Windows 10

**Comment by P0lip** [comment]
@dillonredding 
Hey! The root cause is exactly the same as the one I described in https://github.com/stoplightio/spectral/issues/726.
JS object literals must comply with a certain order of keys as described in the [spec](https://tc39.es/ecma262/#sec-ordinaryownpropertykeys).

**Comment by dillonredding** [comment]
@P0lip 
I was wondering if that might be the case lol. Thanks again for looking into this. Let me know if you come up with a solution!

**Comment by P0lip** [comment]
@dillonredding 
Just wanted to give you heads up that this particular issue is going to be resolved in the next release.

**Comment by dillonredding** [comment]
@P0lip 
Excellent! Do you happen to have a release date yet?

**Comment by P0lip** [comment]
@dillonredding soon! hopefully next week or in 2 weeks :) Def before the end of year.

**Comment by raworre** [comment]
I noticed an issue with getting the `alphabetical` function to work. It seems when a `$ref` exists in the items to be checked for alphabetization the rule is ignored.

## To Reproduce

Using the rules taken directly from [this test scenario](https://github.com/stoplightio/spectral/blob/67faa2a608e794634eea4bc3130048514a5af91e/test-harness/scenarios/alphabetical-responses-order.oas3.scenario) and a modified version of the OpenAPI document in the same test we are not seeing the failed rule as we should.

my-rules.yaml:

```yaml
rules:
  response-order:
    message: Responses should be in alphabetical order
    recommended: true
    given: $.paths.*.*.responses
    then:
      function: alphabetical
```

test-spec.yaml:

```yaml
openapi: 3.0.2
info:
  title: Test Spec
  version: 0.0.0
paths:
  /foo:
    get:
      operationId: get-foo
      responses:
        '404':
          description: ''
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/foo'
components:
  schemas:
    foo:
      title: More incorrect casing
      type: object
      properties:
        id:
          type: integer
        bar:
          type: string
```

CLI command:

```sh
spectral lint test-spec.yaml -r my-rules.yaml
```

We should see the following in our results:

```sh
 19:15  warning  response-order  Responses should be in alphabetical order
```

## Environment

* Library version: 5.0.0
* OS: Windows 10
* Node version: 12.14.1

**Comment by raworre** [comment]
@P0lip does this look like it maybe should be a new issue, or am I missing something with the `alphabetical` implementation?

**Comment by P0lip** [comment]
@raworre it's somewhat related, but a bit different. Could you please file a new issue?
Thank you!

**Comment by raworre** [comment]
@P0lip thanks! Filed [Issue 979](https://github.com/stoplightio/spectral/issues/979).
